### PR TITLE
feat: Import yolo polygons. Includes conversion from bboxes to polygons if user is wishing to transition dataset. 

### PR DESCRIPTION
**Pull request by carbaro**
Hi  

This PR seamlessly integrates with the current yolo.py imports module. 
Based on the yolo_type option, it modifies the content of the task json correspondingly.

Further, it allows:
- datasets split in subdirectories (e.g. train/val with images/labels each)
- prediction confidence values with the option --out_type
- polygonises bboxes automatically
- implements test where test data is polygonised and converted into JSON

### CODE REVIEW

File: label_studio_converter/imports/yolo.py [hunk 1]
```diff
@@ -15,6 +17,23 @@
...
+    image_labels = {}
+    for dir_pth, dir_names, files in os.walk(input_dir):
+        if Path(dir_pth) == Path(input_dir):
+            continue         # skip input_dir. data should be at least one level in from input_dir
+        dir_imgs, dir_lbls = get_images( files ), get_labels( files )
+        if len(dir_imgs) > 0:
+            [images.append(f'{dir_pth}/{img}') for img in dir_imgs]
+        if len(dir_lbls) > 0:
+            [labels.append(f'{dir_pth}/{lbl}') for lbl in dir_lbls]
+    for image, label in zip(images, labels):
```
Comment (obrienwr) [inline label_studio_converter/imports/yolo.py:33]: There's no certainty that the `images` and `labels` lists will be in the same order, something like `for image, label in zip(sorted(images), sorted(labels))` does the trick.

File: label_studio_converter/imports/yolo.py [hunk 2]
```diff
@@ -34,40 +54,68 @@ def convert_yolo_to_ls(
...
-    images_dir = os.path.join(input_dir, 'images')
-    logger.info('Converting labels from %s', labels_dir)
+    # retrieve data (image and label paths). handles datasets with data in subdirectories, e.g. train / val / test
+    images, labels, image_labels = get_data(input_dir, image_ext)
+    logger.info('Converting labels found recursively at %s', input_dir)
+    if yolo_type == 'polygonlabels':
+        # verify if current labels are boxes
+        # scan labels list for first non-empty label, peek contents, determine label type
+        for label in labels:
+            with open(labels[0]) as f:
```
Comment (obrienwr) [inline label_studio_converter/imports/yolo.py:93]: should be `label` not `labels[0]`

File: label_studio_converter/imports/yolo.py [hunk 3]
```diff
@@ -34,40 +54,68 @@ def convert_yolo_to_ls(
...
+                sample_lbl = [line.strip() for line in f.readlines()]
+            if len(sample_lbl) == 0:
+                continue
+            else:
+                break   # non-empty label found
+        logger.info(f'sample label: {sample_lbl}')
+        if len(sample_lbl) < 7: # Polygons expected to consist of 7 items. At least three x,y pairs + class
```
Comment (obrienwr) [inline label_studio_converter/imports/yolo.py:100]: I could be wrong here, but I believe that `sample_lbl` will contain a list of strings, each one being the line of the label file (where lines are separate objects). To test whether the sample label has less than 7 fields you should test something like `if len(sample_lbl[0].split(' ')) < 7`

**Review by obrienwr — COMMENTED** [review]
Didn't get to test this out too much, Just thought I'd leave a few comments from issues that came up while using this. Thanks for the help!

**Comment by makseq** [comment]
After careful consideration, we’ve determined that this is more of an improvement than a critical bug. Additionally, it seems to be an outdated request and hasn’t garnered much interest from the community. For these reasons, we will be closing this issue. We will continue developing the converter library as a part of [Label Studio SDK](https://github.com/HumanSignal/label-studio-sdk).

We appreciate your understanding and encourage you to submit your feedback, questions and suggestions here:
https://github.com/HumanSignal/label-studio-sdk/issues

**Pull request closed without merging**